npm/debugTools/dataplane/converter.go

```diff
@@ -0,0 +1,403 @@
+}
+
+// GetProtobufRulesFromIptable returns a list of protobuf rules from node
+func (c *Converter) GetProtobufRulesFromIptable(tableName string) ([]*pb.RuleResponse, error) {
+	err := c.initConverter()
+	if err != nil {
+		return nil, fmt.Errorf("error occurred during getting protobuf rules from iptables : %w", err)
+	}
+
+	p := &Parser{}
+	ipTableObj := p.ParseIptablesObject(tableName)
+	ruleResList, err := c.pbRuleList(ipTableObj)
+	if err != nil {
+		return nil, fmt.Errorf("error occurred during getting protobuf rules from iptables : %w", err)
+	}
+
+	return ruleResList, nil
+}
+
+// Create a list of protobuf rules from iptable
+func (c *Converter) pbRuleList(ipTableObj *Iptables) ([]*pb.RuleResponse, error) {
+	ruleResList := make([]*pb.RuleResponse, 0)
+	for _, v := range ipTableObj.Chains {
+		chainRules, err := c.getRulesFromChain(v)
+		if err != nil {
+			return nil, fmt.Errorf("error occurred during getting protobuf rule list : %w", err)
+		}
+		ruleResList = append(ruleResList, chainRules...)
+	}
+
+	return ruleResList, nil
+}
+
+func (c *Converter) getRulesFromChain(iptableChainObj *IptablesChain) ([]*pb.RuleResponse, error) {
+	rules := make([]*pb.RuleResponse, 0)
+	for _, v := range iptableChainObj.Rules {
+		rule := &pb.RuleResponse{}
+		rule.Chain = iptableChainObj.Name
+		if _, ok := c.RequiredChainsMap[rule.Chain]; !ok {
+			continue
+		}
+		rule.Protocol = v.Protocol
+		switch v.Target.Name {
+		case "MARK":
+			rule.Allowed = true
+		case "DROP":
+			rule.Allowed = false
+		default:
+			// ignore other targets
+			continue
+		}
+		direction := c.getRuleDirection(iptableChainObj.Name)
+		if direction >= 0 {
+			rule.Direction = direction
+		}
+
+		err := c.getModulesFromRule(v.Modules, rule)
+		if err != nil {
+			return nil, fmt.Errorf("error occurred during getting rules from chain : %w", err)
+		}
+		rules = append(rules, rule)
+	}
+	return rules, nil
+}
+
+func (c *Converter) getRuleDirection(iptableChainObjName string) pb.Direction {
+	if strings.Contains(iptableChainObjName, "EGRESS") {
+		return pb.Direction_EGRESS
+	} else if strings.Contains(iptableChainObjName, "INGRESS") {
+		return pb.Direction_INGRESS
+	}
+	return pb.Direction_UNDEFINED
+}
+
+func (c *Converter) getSetType(name string, m string) pb.SetType {
+	// TODO: Handle CIDR blocks
+	if m == "ListMap" { // labels of namespace
+		if strings.Contains(name, util.IpsetLabelDelimter) {
+			if strings.Count(name, util.IpsetLabelDelimter) > 1 {
+				return pb.SetType_NESTEDLABELOFPOD
+			}
+			return pb.SetType_KEYVALUELABELOFNAMESPACE
+		}
+		return pb.SetType_KEYLABELOFNAMESPACE
+	}
+	if strings.HasPrefix(name, util.NamespacePrefix) {
+		return pb.SetType_NAMESPACE
+	}
+	if strings.HasPrefix(name, util.NamedPortIPSetPrefix) {
+		return pb.SetType_NAMEDPORTS
+	}
+	if strings.Contains(name, util.IpsetLabelDelimter) {
+		return pb.SetType_KEYVALUELABELOFPOD
+	}
+	if matched, _ := regexp.MatchString(`(?i)[^ ]+-in-ns-[^ ]+-\d(out|in)\b`, name); matched {
```

Comment:
it might be more efficient to pre-allocate this static matcher at the package scope with ex: `var matcher = regexp.MustCompile(...)` but I haven't confirmed this